packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -83,15 +88,31 @@ - (void)setRotation:(CLLocationDegrees)rotation {
 }
 
 - (void)setVisible:(BOOL)visible {
-  self.marker.map = visible ? self.mapView : nil;
+  // If marker belongs the cluster manager, visibility need to be controlled with the opacity
+  // as the cluster manager controls when marker is on the map and when not.
+  // Alpha value for marker must always be interpreted before visibility value.
+  if (self.clusterManagerId && self.clusterManagerId != (id)[NSNull null]) {
+    self.marker.opacity = visible ? self.marker.opacity : 0.0f;
+  } else {
+    self.marker.map = visible ? self.mapView : nil;
+  }
 }
 
 - (void)setZIndex:(int)zIndex {
   self.marker.zIndex = zIndex;
 }
 
+- (void)updateMarkerUserData {
+  if (self.clusterManagerId) {
+    self.marker.userData = @[ self.markerId, self.clusterManagerId ];
+  } else {
+    self.marker.userData = @[ self.markerId ];
+  }
+}
```

Comment:
Is there a different way to store these information? Using array as userData is hard to maintain. 
Maybe use a category to add new properties to the GMSMarker class?

Reply:
Created category to handle markerId and clusterManagerId. The store method is still array. Reason for this is that I'm not 100% sure if it would break something if the markerId would not be handled as first element in the array. But now the get and set functions are used to store and read these ids.